packages/flame/lib/src/components/component.dart

```diff
@@ -265,19 +266,16 @@ class Component {
     return (_parent is T ? _parent : _parent?.findParent<T>()) as T?;
   }
 
-  /// Returns the first child that matches the given type [T].
-  ///
-  /// As opposed to `children.whereType<T>().first`, this method returns null
-  /// instead of a [StateError] when no matching children are found.
+  /// Returns the first child that matches the given type [T], or null if there
+  /// are no such children.
   T? firstChild<T extends Component>() {
-    final it = children.whereType<T>().iterator;
-    return it.moveNext() ? it.current : null;
+    return children.whereType<T>().firstOrNull;
```

Comment:
We could make this more efficient if we check whether there is a query registered for `T`

Reply:
That could be on the `ComponentSet`, which would override the `whereType<T>`.

Reply:
True true

Reply:
Opened https://github.com/bluefireteam/ordered_set/issues/20